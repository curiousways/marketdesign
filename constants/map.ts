export const MAP_VIEWBOX_WIDTH = 1016;
export const MAP_VIEWBOX_HEIGHT = 869;
export const MAP_VIEWBOX = `0 0 ${MAP_VIEWBOX_WIDTH} ${MAP_VIEWBOX_HEIGHT}`;
export const MAP_REGION_KEYS: { [key in number]: string } = {
  3: 's1',
  4: 's2',
  5: 's3',
  6: 's4',
  7: 's5',
  21: 'b1',
  22: 'b2',
  23: 'b3',
  24: 'b4',
  25: 'b5',
  26: 'i1',
  27: 'i2',
  28: 'i3',
  29: 'i4',
  30: 'i5',
};
