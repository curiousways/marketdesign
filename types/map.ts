import { RoleId } from './roles';

export type HighlightedMapRegions = Partial<{
  [key in RoleId]: string[];
}>;
