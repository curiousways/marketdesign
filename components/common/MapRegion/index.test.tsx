import { fireEvent, render, screen } from '@testing-library/react';
import { MAP_INDICES } from '../../../constants/map';
import { MAP_REGION_PATHS, MapRegion } from './index';

describe('MapRegion', () => {
  it('returns the transformed path', () => {
    const mapIndex = 12;
    const { container } = render(<MapRegion index={mapIndex} />);

    const path = screen.getByTestId('map-region');

    expect(path.getAttribute('d')).toBe(MAP_REGION_PATHS[mapIndex]);
    expect(container).toMatchSnapshot();
  });

  it('returns the selected path only', () => {
    const mapIndex = 12;

    render(
      <svg>
        <MapRegion pathOnly index={mapIndex} />
      </svg>,
    );

    const path = screen.getByTestId('map-region');

    expect(path.getAttribute('d')).toBe(MAP_REGION_PATHS[mapIndex]);
  });

  it('calls the onClick callback', () => {
    const onClick = jest.fn();

    render(
      <MapRegion roleId="buyer" index={12} region="b1" onClick={onClick} />,
    );

    const path = screen.getByTestId('map-region');

    fireEvent.click(path);

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith('b1', 12);
    expect(path).toHaveClass('cursor-pointer');
  });

  it('does not make the path clickable if no role ID is given', () => {
    const onClick = jest.fn();

    render(<MapRegion index={12} region="b1" onClick={onClick} />);

    const path = screen.getByTestId('map-region');

    fireEvent.click(path);

    expect(onClick).not.toHaveBeenCalled();
    expect(path).not.toHaveClass('cursor-pointer');
  });

  it.each`
    roleId       | expectedColour
    ${'buyer'}   | ${'#9D7F69'}
    ${'seller'}  | ${'#BFDDB3'}
    ${undefined} | ${'white'}
  `(
    'uses $expectedColour as the fill colour for the $roleId role',
    ({ roleId, expectedColour }) => {
      render(<MapRegion index={21} roleId={roleId} />);

      const path = screen.getByTestId('map-region');

      expect(path.getAttribute('fill')).toBe(expectedColour);
    },
  );

  it.each(['woodland', 'wetland'])(
    'renders a region filled with the %s icon',
    (iconType) => {
      render(<MapRegion index={21} roleId="buyer" region={`s1-${iconType}`} />);

      expect(screen.getByTestId('map-region-icon')).toMatchSnapshot();
    },
  );

  it.each(['woodland', 'wetland'])(
    'renders a small region filled with the %s icon',
    (iconType) => {
      render(
        <MapRegion
          isSmall
          size={50}
          index={21}
          roleId="buyer"
          region={`s1-${iconType}`}
        />,
      );

      expect(screen.getByTestId('map-region-icon')).toMatchSnapshot();
    },
  );
});
