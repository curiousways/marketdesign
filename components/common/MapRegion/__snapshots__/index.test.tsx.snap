// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapRegion returns the transformed path 1`] = `
<div>
  <svg>
    <svg
      viewBox="0 0 1016 869"
    >
      <g
        transform="translate(-254, -124.30000000000001) scale(5.08, 5.08)"
      >
        <g>
          <path
            class=""
            d="M350 35.5V101L424 224.5L500.5 186.5L459 138L438.5 149L407.5 61L359 29.5L350 35.5Z"
            data-testid="map-region"
            fill="white"
            stroke="black"
            vector-effect="non-scaling-stroke"
          />
        </g>
      </g>
    </svg>
  </svg>
</div>
`;
